### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by aliabbani**
## OOP school library: set up associations

In this milestone I:

- Created a class Classroom with the following:
@label instance variable should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
- Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
Create the has-many side (a classroom has many students).
Create the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
- Created a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
Create the has-many side of Book and Rental (a book has many rentals).
Create the belongs-to side of Rental and Book (a rental belongs to a book).
Create the has-many side of Person and Rental (a person has many rentals).
Create the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
```
Comment (eliasecasta) [inline book.rb]: - [x] Getters and setters for rentals should be present, please do so to comply with the project requirements.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (eliasecasta) [inline book.rb:8]: - [x]  A method for adding rentals should be present. Link to the example in a previous lesson: [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). Please add it to comply with the project requirements.

File: person.rb [hunk 3]
```diff
@@ -1,15 +1,16 @@
-require './corrector.rb'
+require './corrector'
 
 class Person
   attr_accessor :name, :age, :parent_permission
-  attr_reader :id, :corrector
+  attr_reader :id, :corrector, :rentals
```
Comment (eliasecasta) [inline person.rb]: - [x] Getters and Setters for rentals should be present, please do so to comply with the project requirements (add it to the accessor part of this file)

File: person.rb [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
 
-  def initialize(age, name = 'unknown', parent_permission: true)
+  def initialize(age, name = 'unknown', parent_permission = { parent_permission: true })
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
     @corrector = Corrector.new
+    @rentals = []
   end
```
Comment (eliasecasta) [inline person.rb:15]: - [x] A method for adding rentals should be present. Link to the example in a previous lesson: [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). Please do so to comply with the project requirements.

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @aliabbani, I am Elias, your reviewer for this PR

Well done so far with your project!

Your project is almost correctly set, you missed some really small details, keep it up!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [ ] Please add a descriptive Readme to your project to comply with the project requirements.
- [ ] Please check comments below for extra requirements.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (aliabbani) [reply book.rb]: Thank you for your valuable comment. I changed it as you can see in the image below:
attr2

File: book.rb (hunk 2 above)
Comment (aliabbani) [reply book.rb:8]: Thank you for your valuable comment. I added the add_rental method as you can see in the image below:

method2

File: person.rb (hunk 3 above)
Comment (aliabbani) [reply person.rb]: Thank you for your valuable comment. I changed it as you can see in the image below:
personattr

File: person.rb (hunk 4 above)
Comment (aliabbani) [reply person.rb:15]: Thank you for your valuable comment. I added the add_rental method as you can see in the image below:
rentalinpersonmethod

**Review by mikethreels — APPROVED** [review]
Hi Abbani, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**